libraries/AP_WindVane/AP_WindVane.cpp

```diff
@@ -190,47 +190,45 @@ void AP_WindVane::init(const AP_SerialManager& serial_manager)
 
     // wind direction
     switch (_direction_type) {
-        case WindVaneType::WINDVANE_NONE:
-            // WindVane disabled
-            return;
         case WindVaneType::WINDVANE_HOME_HEADING:
         case WindVaneType::WINDVANE_PWM_PIN:
             _direction_driver = new AP_WindVane_Home(*this);
             break;
         case WindVaneType::WINDVANE_ANALOG_PIN:
             _direction_driver = new AP_WindVane_Analog(*this);
             break;
-        case WindVaneType::WINDVANE_SITL:
 #if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+        case WindVaneType::WINDVANE_SITL:
             _direction_driver = new AP_WindVane_SITL(*this);
-#endif
             break;
+#endif
         case WindVaneType::WINDVANE_NMEA:
             _direction_driver = new AP_WindVane_NMEA(*this);
             _direction_driver->init(serial_manager);
             break;
+        default:
```

Comment:
So now when someone adds a new windvane type the compiler will not catch them and tell them they need to add something to this block.  I don't think that's an improvement.